astropy/coordinates/representation.py

```diff
@@ -2066,6 +2133,31 @@ def from_cartesian(cls, cart):
 
         return cls(phi=phi, theta=theta, r=r, copy=False)
 
+    def transform(self, matrix):
+        r"""Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+
+        """
+        # TODO! differentials
+        xyz = erfa_ufunc.s2c(self.phi, 90*u.deg-self.theta)
+        p = erfa_ufunc.rxp(matrix, xyz)
+        lon, lat, ur = erfa_ufunc.p2s(p)
+
+        # construct in original units
```

Comment:
As noted above, I'd not do that - it does cause an extra overhead and I don't think that is worth it. (I do think representations should gain a `to(unit)` method, with a single unit just for that of the distance/radius, and maybe named arguments for the angles, etc.  But for another PR!).